paf-mvp-frontend/src/lib/paf-lib.ts

```diff
@@ -581,30 +584,128 @@ export const getIdsAndPreferences = (): IdsAndPreferences | undefined => {
 };
 
 /**
+ * Aggregate Seed and Preferences for storage.
  *
- * @param proxyHostName
- * @param transactionIds
+ * We must record the Ids and Preferences
+ * in case they change in the cookies after the
+ * generation of the Seed.
  */
+export interface SeedEntry {
+  seed: Seed;
+  idsAndPreferences: IdsAndPreferences;
+}
+
+type PrebidTransactionId = string;
+type SlotId = string;
+type ContentId = string;
+
+export interface TransmissionRegistryContext {
+  /** Transaction Id generated by Prebidjs. */
+  prebidTransactionId: PrebidTransactionId;
+  /** Transaction Id generated for the PAF Audit Log and used for signing the Seed. */
+  pafTransactionId: TransactionId;
+  slotId: SlotId;
+  contentId: ContentId;
+}
+
+/**
+ * Internal storage for generated Seeds.
+ */
+const seedStorage = new Map<TransactionId, SeedEntry>();
+const auditLogByPrebidTransactionId = new Map<PrebidTransactionId, AuditLog>();
+const prebidTransactionIdBySlotId = new Map<SlotId, PrebidTransactionId>();
+
 export const createSeed = async (
```

Comment:
shouldn't we stop exporting this `createSeed` method?
I feel like this is becoming private stuff.

Reply:
(this would mean to unit-test `generateSeed`, which seems legit)

Reply:
It makes sense. I merge and do a PR for that.